### PR TITLE
Descendents at depth.

### LINKED ISSUE #23: Allow to find descendents at a certain depth.
Currently findDescendents() searches down to a bottom depth, and I'd also like to be able to pass a top depth.

This is not critical functionality for me, but I think it would be a nice feature. I'm about to do a patch on the drupal module that would use this feature if it was available.

My proposal i don't think would break any existing functionality. But I could also create a new method `findDescendentsAtDepth` 

```
  public function findDescendants(NodeKey $nodeKey, $depth = 0) {
```
becomes 
```
  public function findDescendants(NodeKey $nodeKey, $end_depth = 0, $start_depth = 0) {
```

### PR DESCRIPTION
**Pull request by simesy**
Without disturbing the existing $depth parameter, this provides an additional $start parameter. This is useful for those wanting to process a tree in layers (say building a visual representation), or if there is meaning to a specific layer, without doing a lot of php processing of the result set.

With the tree used in the tests, you can for example take 1 depth starting at depth 2. 

```
                    (1)       <-----------------------------+
                   1 | 22                          Root node.
                     |
            ————————————————————————
           (2)                    (3)
          2 | 9                 10 | 21
            |                      |
            |                      |
    —————————          ————————————————————
   (4)                (7)       (8)      (9)     <----------+
  3 | 8             11 | 16    17 18    19 20       $depth 1.
    |                  |                            $start 2.
    |                  |
 ————————         ———————————
(5)    (6)      (10)       (11)
4 5    6 7     12  13     14  15
```

Closes #23

### CODE REVIEW

File: src/NestedSetInterface.php [hunk 1]
```diff
@@ -80,11 +80,13 @@ public function deleteSubTree(Node $node);
...
    *   (optional) A depth limit. Defaults to 0, no limit.
+   * @param int $start
+   *   (optional) A starting depth. Default of 1 excludes the node itself.
    *
    * @return array
-   *    The nested array of descendants.
+   *   The nested array of descendants.
    */
-  public function findDescendants(NodeKey $nodeKey, $depth = 0);
+  public function findDescendants(NodeKey $nodeKey, $depth = 0, $start = 0);
```
Comment (kimpepper) [inline src/NestedSetInterface.php]: but this defaults to 0.

**Review by kimpepper — CHANGES_REQUESTED** [review]

File: src/Storage/DbalNestedSet.php [hunk 2]
```diff
@@ -129,7 +128,7 @@ protected function doInsertNode(NodeKey $nodeKey, $left, $right, $depth) {
   /**
    * {@inheritdoc}
    */
-  public function findDescendants(NodeKey $nodeKey, $depth = 0) {
+  public function findDescendants(NodeKey $nodeKey, $depth = 0, $start = 1) {
```
Comment (kimpepper) [inline src/Storage/DbalNestedSet.php:131]: and this defaults to 1

File: tests/Functional/DbalNestedSetTest.php [hunk 3]
```diff
@@ -616,6 +662,23 @@ public function testGetTree() {
 
   /**
    * Loads the test data.
+   *
```
Comment (larowlan) [inline tests/Functional/DbalNestedSetTest.php:665]: ❤️

**Review by larowlan — COMMENTED** [review]
looks good to me

File: src/NestedSetInterface.php (hunk 1 above)
Comment (simesy) [reply src/NestedSetInterface.php]: Oh it should be 1. Thanks!

Comment (simesy) [reply src/NestedSetInterface.php]: Pushed the fix.

**Pull request merged**